lib/sqlalchemy/ext/horizontal_shard.py

```diff
@@ -176,9 +233,13 @@ def _choose_shard_and_assign(self, mapper, instance, **kw):
             state.identity_token = shard_id
         return shard_id
 
-    def connection_callable(
-        self, mapper=None, instance=None, shard_id=None, **kwargs
-    ):
+    def connection_callable(  # type: ignore [override]
```

Comment:
I get `error: Signature of "connection_callable" incompatible with supertype "Session"  [override]` even if I make the signature exactly the same as specified by `_ConnectionCallableProto` in `lib/sqlalchemy/orm/session.py`.